README.md

```diff
@@ -250,12 +247,12 @@ push:
 
 See the [push-on-changes.yaml](examples/push-on-changes/buf-ci.yaml) example.
 
+### Verify generated files are up-to-date
 
-### Check generation
+If your project uses local code generation, we recommend checking for diffs on pull requests (and failing if there is a diff).
+This isn't available as a built-in step because generating code may require manual setup, but can easily be added by invoking `buf`.
```

Comment:
Might be valuable to add this in as a step thats off by default. I do think a lot of users would want this behaviour, but easy to add later with feedback.

Reply:
Behaviour could be gated by the existence of a `buf.gen.yaml` file.

Reply:
Yeah, I think it gets tricky because we don't know in general if it will succeed or not. If there is a way we could do it sanely, we should definitely consider. I don't know that there is though.